packages/core/contracts/financial-templates/common/FundingRateApplier.sol

```diff
@@ -141,40 +131,39 @@ contract FundingRateApplier is Lockable {
             currentTime,
             paymentPeriod,
             _latestFundingRatePerSecond.rawValue,
-            effectiveFundingRateForPeriod.rawValue
+            periodRate.rawValue
         );
 
         lastUpdateTime = currentTime;
     }
 
     function _calculateEffectiveFundingRate(
         uint256 paymentPeriodSeconds,
-        FixedPoint.Unsigned memory fundingRatePerSecond,
-        FixedPoint.Unsigned memory feeMultiplier
-    ) internal pure returns (FixedPoint.Unsigned memory, FixedPoint.Unsigned memory) {
+        FixedPoint.Signed memory fundingRatePerSecond,
+        FixedPoint.Unsigned memory currentCumulativeFundingRateMultiplier
+    )
+        internal
+        pure
+        returns (FixedPoint.Unsigned memory newCumulativeFundingRateMultiplier, FixedPoint.Signed memory periodRate)
+    {
+        // Note: this method uses named return variables to save a little bytecode.
+
         // Determine whether `fundingRatePerSecond` implies a negative or positive funding rate,
         // and apply it over a pay period.
-        FixedPoint.Unsigned memory ONE = FixedPoint.fromUnscaledUint(1);
-        FixedPoint.Unsigned memory effectiveFundingRateForPeriod = ONE;
-
-        // If `fundingRatePerSecond` == 1, then maintain the current multiplier.
-        FixedPoint.Unsigned memory newFeeMultiplier = feeMultiplier;
-        if (fundingRatePerSecond.isGreaterThan(ONE)) {
-            // If `fundingRatePerSecond` > 1, then first scale the funding over the pay period:
-            // (`fundingRatePerSecond` - 1) * payPeriod = effectiveFundingRate.
-            // Next, multiply the current multipier by (1 + effectiveFundingRate).
-
-            effectiveFundingRateForPeriod = ONE.add(fundingRatePerSecond.sub(ONE).mul(paymentPeriodSeconds));
-            newFeeMultiplier = feeMultiplier.mul(effectiveFundingRateForPeriod);
-        } else if (fundingRatePerSecond.isLessThan(ONE)) {
-            // If `fundingRatePerSecond` < 1, then first scale the funding over the pay period:
-            // (1 - `fundingRatePerSecond`) * payPeriod = effectiveFundingRate.
-            // Next, multiply the current multipier by (1 - effectiveFundingRate).
-
-            effectiveFundingRateForPeriod = ONE.sub(ONE.sub(fundingRatePerSecond).mul(paymentPeriodSeconds));
-            newFeeMultiplier = feeMultiplier.mul(effectiveFundingRateForPeriod);
-        }
-
-        return (newFeeMultiplier, effectiveFundingRateForPeriod);
+        FixedPoint.Signed memory ONE = FixedPoint.fromUnscaledInt(1);
+
+        // Multiply the per-second rate over the number of seconds that have elapsed to get the period rate.
+        periodRate = fundingRatePerSecond.mul(SafeCast.toInt256(paymentPeriodSeconds));
+
+        // Add one to create the multiplier to scale the existing fee multiplier.
+        FixedPoint.Signed memory signedPeriodMultiplier = ONE.add(periodRate);
```

Comment:
nit: is there any point in declaring this `ONE` variable now that it only used in one place?

Reply:
Nice catch this might save some bytes!

Reply:
Good point -- will try removing it.

Reply:
Removing the `ONE` seems to _add_ 10 bytes. Should we still remove it or keep it?

Reply:
I'd still remove it. we dont declare scalers like this anywhere else. Also every byte counts ;)